docs/modules/ROOT/pages/dapp.adoc

```diff
@@ -0,0 +1,337 @@
+= Building a Dapp with Solidity Hot Loader
+
+== Introduction
+
+In this guide, we will build a simple Dapp using the Solidity Hot Loader. We will create a Counter Dapp, which displays the value of a Counter that can be increased. We will update the code in the smart contract to show the power of the Solidity Hot Loader to automatically update the Dapp.
+
+== Setup
+
+We will use the https://create-react-app.dev/[`create-react-app`] package to bootstrap a React application.
+
+First we need a local blockchain to develop on. We use `ganache-cli`, so go ahead and install (if you haven’t already) and then run a local blockchain using `ganache-cli`.
+
+[source,console]
+----
+$ npm install -g ganache-cli
+$ ganache-cli --deterministic
+----
+
+Next we need a project directory to develop our project in. In a separate terminal create the project directory and create `package.json` with `npm init`.
+
+[source,console]
+----
+$ mkdir myproject && cd myproject
+$ npm init -y
+----
+
+We use the OpenZeppelin CLI for compiling and deploying our contract, so go ahead and install *OpenZeppelin CLI*.
+
+[source,console]
+----
+$ npm install @openzeppelin/cli
+----
+
+Next, initialize the project with OpenZeppelin. This will create directories and config for our project.
+
+[source,console]
+----
+$ npx openzeppelin init
+? Welcome to the OpenZeppelin SDK! Choose a name for your project myproject
+? Initial project version 1.0.0
+Project initialized. Write a new contract in the contracts folder and run 'openzeppelin create' to deploy it.
+----
+
+== Counter contract
+
+Our Counter smart contract consists of an integer `value` and a function to increase it by one.
+
+In the `contracts` directory save `Counter.sol` with the following contents:
+
+[source,solidity]
+----
+pragma solidity ^0.5.0;
+
+contract Counter {
+    uint256 public value;
+
+    function increase() public {
+        value = value + 1;
+    }
+}
+----
+
+Next we deploy the Counter contract to our local blockchain.
+
+Run `npx openzeppelin create` and use the interactive commands to deploy the Counter contract to the development network. There is no function that we need to call to `initialize`.
+
+[source,console]
+----
+$ npx openzeppelin create
+✓ Compiled contracts with solc 0.5.14 (commit.1f1aaa4)
+? Pick a contract to instantiate Counter
+? Pick a network development
+✓ Added contract Counter
+✓ Contract Counter deployed
+All contracts have been deployed
+? Call a function to initialize the instance after creating it? No
+✓ Setting everything up to create contract instances
+✓ Instance created at 0xCfEB869F69431e42cdB54A4F4f105C19C080A601
+0xCfEB869F69431e42cdB54A4F4f105C19C080A601
+----
+
+== Creating the dapp
```

Comment:
```suggestion
== Creating the Dapp
```
I use https://capitalizemytitle.com/ to check for title capitalization, the 'rules' are not very straightforward